src/python/grpcio_tests/tests/testing/_connection_test.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import unittest
+
+from grpc.framework.foundation import logging_pool
+import grpc_testing
+
+from tests.testing import _application_testing_common
+from tests.testing import _client_application
+from tests.testing import _server_application
+
+
+class ConnectionTest(unittest.TestCase):
+
+    def setUp(self):
+        self._application_thread_pool = logging_pool.pool(1)
+        self._fake_time = grpc_testing.fake_time(time.time())
+        self._real_time = grpc_testing.real_time()
+        self._fake_time_connection = (grpc_testing.connection_from_descriptions(
+            {
+                _server_application.FirstServiceServicer():
+                _application_testing_common.FIRST_SERVICE
+            }, self._fake_time))
+        self._real_time_connection = (grpc_testing.connection_from_descriptions(
+            {
+                _server_application.FirstServiceServicer():
+                _application_testing_common.FIRST_SERVICE
+            }, self._real_time))
+
+    def tearDown(self):
+        self._application_thread_pool.shutdown(wait=True)
+
+    def test_successful_unary_unary(self):
+        application_future = self._application_thread_pool.submit(
+            _client_application.run, _client_application.Scenario.UNARY_UNARY,
+            self._fake_time_connection.channel())
+        application_return_value = application_future.result()
```

Comment:
[Testing orthodoxy](http://wiki.c2.com/?ArrangeActAssert): accessing the application future's result is part of the test action (because it's not just a simple getter but rather something that blocks until the system under test has returned and raises an exception if the system under test raised an exception).

Once the value has been obtained, the test action is over and the test assertions happen after the blank line.